rEXIFier.sh

```diff
@@ -783,10 +773,51 @@ for WIPSortedFileAbsolutePath in ${WIPSortedFileAbsolutePaths[@]}; do
+                  fi
+                # If the duplicate file exists, run deduplication process (the filename will be amended by adding an index)
                 else
-                  printf "Moving ${DestinationDuplicateFile} to '${DestinationPath}/${FileContentDuplicates}/${WIPDirectoryName}/' ${NOTE} SUCCESSFUL\n"
+                  # Search for file basename match
+                  MatchedFileContentDuplicates=$(find ${DestinationPath}/${FileContentDuplicates}/${WIPDirectoryName} -maxdepth 1 -type f -iname "${FormatedFileNameNoExtension}*.${NormalisedFileExtension}" )
+                  # Sorting file names to process them sequentially
+                  FileListSorter ${MatchedFileContentDuplicates}
+                  # Returning the value from FileListSorter function by assigning the value of output (SortedFileList) to an array
+                  # Bash functions, unlike functions in most programming languages do not allow you to return values to the caller
+                  SortedMatchedFileContentDuplicates=${SortedFileList}
+                  # Resetting SortedFileList array for sanity
+                  SortedFileList=""
+                  # Getting the last member in the sorted array of matched filenames, mind the difference in the construct 
+                  # ${var[@]-1} finds the last element, versus ${var[@] -1} finds the last character of the last element
+                  # ${var[*]-1} and ${var[*] -1} perform as above
+                  LastSortedMatchedFileContentDuplicate=${SortedMatchedFileContentDuplicates[@]:-1}
+                  # Extracting file basename
+                  LastSortedMatchedFileDuplicateBasename=$(basename ${LastSortedMatchedFileContentDuplicate})
+                  # Extracting file name
+                  LastSortedMatchedFileContentDuplicate=${LastSortedMatchedFileDuplicateBasename%.*} 
+                  # Extracting the duplicate index
+                  LastSortedMatchedFileDuplicateIndex=$(echo ${LastSortedMatchedFileNameDuplicate} | awk -F "-" '{print $5}')
+                  # Setting the value if the filename does not contain a duplicate index
+                  if [[ LastSortedMatchedFileDuplicateIndex == "" ]] ; then
+                    LastSortedMatchedFileDuplicateIndex="00"
+                  fi
+                  # Incrementing the duplicate index so that we can form a new File Name
+                  printf -v IncrementedLastSortedMatchedFileDuplicateIndex %02d "$((10#$LastSortedMatchedFileDuplicateIndex + 1))"
+                  # Passing the incremented duplicate index to IncrementedDuplicateIndex
+                  IncrementedDuplicateIndex=${IncrementedLastSortedMatchedFileDuplicateIndex}
+                  # Re-forming the File Name with the incremented duplicate index
+                  FormatedFileName=${FormatedFileNameNoExtension}-${IncrementedDuplicateIndex}.${NormalisedFileExtension}
```

Comment:
this whole block of code does the same thing as above, would be better to move to a reusable function